test/CadlRanchProjectsNonAzure.Tests/type-union.cs

```diff
@@ -0,0 +1,197 @@
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Scm._Type.Union;
+using Scm._Type.Union.Models;
+
+using AutoRest.TestServer.Tests.Infrastructure;
+using NUnit.Framework;
+
+namespace CadlRanchProjectsNonAzure.Tests
+{
+    public class TypeUnionTests : CadlRanchNonAzureTestBase
+    {
+        [Test]
+        public Task GetStringsOnly() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetStringsOnlyClient().GetStringsOnlyAsync();
+            Assert.AreEqual(200, response.GetRawResponse().Status);
+            Assert.AreEqual(GetResponse9Prop.B, response.Value.Prop);
+        });
+
+        [Test]
+        public Task SendStringsOnly() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetStringsOnlyClient().SendAsync(SendRequest9Prop.B);
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task GetStringExtensibleOnly() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetStringExtensibleClient().GetStringExtensibleAsync();
+            Assert.AreEqual(200, response.GetRawResponse().Status);
+            Assert.AreEqual(new GetResponseProp3("custom"), response.Value.Prop);
+        });
+
+        [Test]
+        public Task SendStringExtensibleOnly() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetStringExtensibleClient().SendAsync(new GetResponseProp3("custom"));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task GetStringExtensibleNamedOnly() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetStringExtensibleNamedClient().GetStringExtensibleNamedAsync();
+            Assert.AreEqual(200, response.GetRawResponse().Status);
+            Assert.AreEqual(new StringExtensibleNamedUnion("custom"), response.Value.Prop);
+        });
+
+        [Test]
+        public Task SendStringExtensibleNamedOnly() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetStringExtensibleNamedClient().SendAsync(new StringExtensibleNamedUnion("custom"));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task GetIntsOnly() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetIntsOnlyClient().GetIntsOnlyAsync();
+            Assert.AreEqual(200, response.GetRawResponse().Status);
+            AssertEqual(BinaryData.FromObjectAsJson(2), response.Value.Prop);
+        });
+
+        [Test]
+        public Task SendIntsOnly() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetIntsOnlyClient().SendAsync(BinaryData.FromString("2"));
```

Comment:
well this is misleading because I checked how this is defined, the API is expecting the number `2` but we are constructing from string `2`
```suggestion
            var response = await new UnionClient(host, null).GetIntsOnlyClient().SendAsync(BinaryData.FromObjectAsJson(2));
```

Reply:
Please sync the change to the azure version test case as well

Reply:
fixed